index.html

```diff
@@ -767,15 +768,10 @@ <h3>
               </li>
             </ol>
           </li>
+          <li>Set |lock|'s {{[[Released]]}} internal slot to `true`.
+          </li>
           <li>
-            <a>Queue a task</a> to run the following steps:
-            <ol>
-              <li>Set |lock|'s {{[[Released]]}} internal slot to `true`.
-              </li>
-              <li>
-                <a>Fire an event</a> named "`release`" at |lock|.
-              </li>
-            </ol>
+            <a>Fire an event</a> named "`release`" at |lock|.
```

Comment:
Seems odd that this is not queued? or was it queued above and I missed it?

Reply:
Heh, that's how this entire PR started: I asked in #293 whether it made sense to queue the task, and we uncovered a lot of issues with "in parallel" in the spec. I also asked above and it seemed like there was no need to queue this at all since we're not running in parallel.

Reply:
So not queuing this means that the following code should work:

```js
let fired = false;
lock.addEventListener("release", () => fired = true);
lock.release();
console.assert(fired === true);
```

Are you sure that's how things work? It seems possible, but I just wanted to check.

Reply:
I _think_ so? I'm honestly fine either way, it's just no clear to me if there's a guideline for deciding whether to queue a task or not, and the fact that `WakeLockSentinel.release()` returns a promise also makes it more confusing:
* Should the snippet above work, or should one have to wait for the promise to resolve?
* Doesn't queuing a task mean it will be fired after the promise resolves?
* If the task is queued, it will likely not be run when the page is being unloaded (as part of the "handling loss of page visibility" section), while if we run these steps directly here they will. Is there a preference?

Reply:
I mean, can you write a test?

Really it seems like `release()` shouldn't return a promise at all, since it's not giving any meaningful information. I wonder if that would be a backward-compatible change.

Reply:
I agree that there is no value in `release()` returning a `Promise` but it would not be a backwards-compatible change since `lock.release().then()` would throw an error if it returned `undefined`.

Reply:
It might be worth adding a note to the spec saying that the returned promise is a historical artifact and only retained for backward compatibility.

Reply:
One thing we can also do is make this part of the change adhere to ["Events should fire before Promises resolve"](https://w3ctag.github.io/design-principles/#promises-and-events) and only resolve the promise after firing the event. The existing version of the spec kind of attempts to do that, but since we queue a task while already in parallel to fire the event I'm not sure this is really guaranteed to happen.

In any case: is there a guideline for deciding whether to queue a task or not when firing an event inside the main event loop?

Reply:
You generally don't queue a task for synchronous things, like this one.

I don't think the guidance is applicable in this case since it's a synchronous thing that only returns a promise for historical reasons.

Reply:
If you fire the event and resolve the promise in the same task, the event will fire before the promise resolves, right? So if you do that in the task that invokes `release()` (i.e., `release()` doesn't queue a task itself and just fires the event and resolves the promise) I think you would be adhering to that, even if slightly weird and probably not to be repeated.

Reply:
Yeah, in which case the current patch would be fine as-is. I've read #226 and #237 again to see if there was any explanation for making `release()` return a Promise, but couldn't find anything other than the suggested Web IDL snippets from #226. It looks weird right now, but I guess it also allows us to do more things in the future if necessary and use it to signal everything has finished running?

Reply:
Yeah, maybe.

Reply:
Thanks for the clarifications on this. Makes sense now. @rakuco, feel free to resolve this if you are happy.